pkg/streamingpromql/aggregations/stddev.go

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/promql/engine.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors
+
+package aggregations
+
+import (
+	"math"
+
+	"github.com/prometheus/prometheus/promql"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/functions"
+	"github.com/grafana/mimir/pkg/streamingpromql/limiting"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+type StddevAggregationGroup struct {
+	floats     []float64
+	floatMeans []float64
+
+	// Keeps track of how many series we have encountered thus far for the group at this point
+	// This is necessary to do per point (instead of just counting the groups) as a series may have
+	// stale or non-existent values that are not added towards the count.
+	// We use float64 instead of uint64 so that we can reuse the float pool and don't have to retype on each division.
```

Comment:
[nit] This might make this a little clearer (same applies for comment in `avg.go` above):

```suggestion
	// Keeps track of how many samples we have encountered thus far for the group at this point
	// This is necessary to do per point (instead of just counting the input series) as a series may have
	// stale or non-existent values that are not added towards the count.
	// We use float64 instead of uint64 so that we can reuse the float pool and don't have to retype on each division.
```